slam-libraries/viam-orb-slam3/orbslam_server_v1.h

```diff
@@ -58,6 +60,8 @@ class SLAMServiceImpl final : public SLAMService::Service {
     double yamlTime;
     std::atomic<bool> offlineFlag{false};
     bool local_viewer_flag = false;
+    bool pure_localization_mode = false;
+    int n_key_frames = 0;
```

Comment:
[q] Is `n_key_frames` something that is save to be and/or should be a global variable within the slam service? Is `n_key_frames` not used to count the number of keyframes in each submap depending on the map you're looking at? If I'm wrong, what does it stand for?

Reply:
number of keyframes will change when you have a valid map update occur that introduces a new key frame. This will cause us to grab all the map points again so we have the most up-to-date map.

If delocalization occurs or a new map is transferred to, this will also change the number of keyframes and cause us to grab a the new data. However I did just realize that if the new map has the same number of key frames the grabbing the new map data would not occur. This was a problem with the past method as well so I have added an additional check for if the current map id has changed as well